test/functional/invalidateblock.py

```diff
@@ -8,19 +8,15 @@
 from test_framework.util import *
 
 class InvalidateTest(BitcoinTestFramework):
-    
-        
+
     def __init__(self):
         super().__init__()
         self.setup_clean_chain = True
         self.num_nodes = 3
 
     def setup_network(self):
-        self.nodes = []
-        self.nodes.append(start_node(0, self.options.tmpdir))
-        self.nodes.append(start_node(1, self.options.tmpdir))
-        self.nodes.append(start_node(2, self.options.tmpdir))
-        
+        self.setup_nodes()
```

Comment:
`setup_nodes()` starts the nodes, but doesn't add connections between them. That's what we need here.